### PR TITLE
test(e2e): deploy eigen core and avs

### PR DESCRIPTION
**Pull request by kevinhalliday**
Deploy eigenlayer core and avs contracts in e2e 

task: none

### CODE REVIEW

File: test/e2e/netman/helpers.go [hunk 1]
```diff
@@ -143,6 +143,45 @@ func deployPortal(ctx context.Context, txOpts *bind.TransactOpts, client *ethcli
 	return proxy, nil
 }
 
+func deployOmniAVS(ctx context.Context, client *ethclient.Client, txOpts *bind.TransactOpts,
+	proxyAdmin common.Address, owner common.Address, portal common.Address, omniChainID uint64,
+	delegationManager common.Address, avsDirectory common.Address, minimumOperatorStake *big.Int,
+	maximumOperatorCount uint32, strategyParams []bindings.IOmniAVSStrategyParams,
```
Comment (kevinhalliday) [inline test/e2e/netman/helpers.go:149]: This arg list is comically long. Any suggestions?

**Review by corverroos — COMMENTED** [review]
I've pushed a PR that decouples the AVS xdapp from netman.

File: test/e2e/netman/manager.go [hunk 2]
```diff
@@ -48,9 +48,15 @@ type Manager interface {
 	// DeployPrivatePortals deploys portals to private (docker) chains.
 	DeployPrivatePortals(ctx context.Context, valSetID uint64, validators []bindings.Validator) error
 
+	// DeployAVS deploys the AVS contract to its configured chain.
+	DeployAVS(ctx context.Context, cfg types.AVSConfig, eigen types.EigenLayerDeployments) error
```
[the remark this answers is not in the document]
Comment (kevinhalliday) [reply test/e2e/netman/manager.go]: Yeah I like that

File: test/e2e/cmd/flags.go [hunk 3]
```diff
@@ -26,3 +27,8 @@ func bindPromFlags(flags *pflag.FlagSet, cfg *app.PromSecrets) {
 	flags.StringVar(&cfg.User, "prom-user", cfg.User, "prometheus user")
 	flags.StringVar(&cfg.Pass, "prom-password", cfg.Pass, "prometheus password")
 }
+
+func bindDeployFlags(flags *pflag.FlagSet, cfg *app.DeployConfig) {
+	bindPromFlags(flags, &cfg.PromSecrets)
+	flags.StringVar(&cfg.EigenFile, "eigen-file", cfg.EigenFile, "path to json file defining eigenlayer deployments. Empty to skip AVS deployment")
+}
```
[the remark this answers is not in the document]
Comment (kevinhalliday) [reply test/e2e/cmd/flags.go:34]: I just used `deployConfig` in base e2e command

**Review by ttarsi — APPROVED** [review]
Exciting let's get e2e tests working before merging though – looks like messaging is still only working for 2 chains

**Pull request merged**